Tactics/Sym.lean

```diff
@@ -23,3 +25,158 @@ macro_rules
       -- (try simp_all (config := {decide := true, ground := true}) only);
       -- (try simp only [ne_eq, r_of_w_different, r_of_w_same, w_of_w_shadow, w_irrelevant])
       (try simp_all (config := {decide := true, ground := true})))
+
+
+theorem run_onestep (s s': ArmState) (n: ℕ) (h_nonneg: 0 < n):
+  (s' = run n s) ↔ ∃ s'', s'' = stepi s ∧ s' = run (n-1) s'' := by
+  cases n
+  · cases h_nonneg
+  · rename_i n
+    simp [run]
+
+-- TODO: replace this with an upcoming new lemma in Std
+theorem Std.BitVec.foldCtor : { toFin := { val := a, isLt := h } : BitVec n } = BitVec.ofNat n a := by
+  simp [BitVec.ofNat, Fin.ofNat', h, Nat.mod_eq_of_lt]
+
+theorem write_mem_bytes_program {n:ℕ} (addr:BitVec 64) (bytes:BitVec (n * 8)):
```

Comment:
`write_mem_bytes_program` would fit better in `Arm/Memory.lean` along with other similar lemmas. See, e.g., https://github.com/leanprover/LNSym/blob/16b24c69e0e93ac2f2e9dae828283f952e003bb9/Arm/Memory.lean#L74.

Reply:
Another minor thing: for uniformity, it'd be nice to have a space between the var and its type. E.g.: `(addr : BitVec 64)` instead of `(addr:BitVec 64)`.